### PR TITLE
Fix fetchAdd and create a test to validate it.

### PR DESCRIPTION
**Pull request by dmk42**
This fixes the fetchAdd and fetchSub operations on reals when atomic intrinsics are used, and adds a regression test.

### CODE REVIEW

File: test/parallel/atomic/dmk/fetchAdd.chpl [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+var n: atomic int;
+var x: atomic real;
+
+n.write(42);
+x.write(3.14);
+
+writeln("old values = ", n, ", ", x);
+writeln("returned values (should equal old values) = ",
+	n.fetchAdd(47), ", ", x.fetchAdd(3.14));
```
Comment (ronawho) [inline test/parallel/atomic/dmk/fetchAdd.chpl]: Can you add a fetchSub check to this test too?

**Comment by mppf** [comment]
Looks good to me. As a side note, I personally tend to believe the Linux Kernel community's arguments that 'volatile' is almost never useful. In particular I don't think it's useful in the code here. It's the full_memory_barrier() and/or the  my__sync_bool_compare_and_swap() that prevent reorderings of the access to those variables.

File: test/parallel/atomic/dmk/fetchAdd.chpl (hunk 1 above)
Comment (dmk42) [inline test/parallel/atomic/dmk/fetchAdd.chpl]: Done.

**Comment by dmk42** [comment]
Any final comments now that I've updated the regression test to cover both fetchAdd and fetchSub?

**Comment by ronawho** [comment]
Nope, good to merge

**Pull request merged**